_posts/2024-07-30-configuration-management.markdown

```diff
@@ -0,0 +1,41 @@
+---
+layout: post
+title:  "Safe and Secure Configuration Management on CHERIoT"
+date:   2024-07-30
+categories: security philosophy
+author: Phil Day
+---
+All systems rely on some form of configuration data.
+Fully immutable systems, where configuration is baked in at build time, may work for container based environments where re-deployment is relatively easy, but in other systems the configuration interface forms a significant part of the attack surface, and misconfiguration, whether accidental or malicious, is a major sources of security vulnerabilities.
```

Comment:
It might be nice to mention here that the ClownStrike problem was caused by a configuration data update.  This is particularly relevant: they had an in-kernel parser with a bug, which crashed the kernel on the new configuration update.  With your design, it would have crashed in the validator and just rejected the configuration change with no impact on the wider system.